### PR TITLE
chore: implement migration guide for seamless user  transition from l…

### PR DESCRIPTION
**Pull request by 1makedadavis**
…egacy code to new SDK

## Description

Implemented Migration guide for seamless user transition from legacy code to new Typescript SDK

## Story

[JIRA-726](https://lobsters.atlassian.net/browse/DXP-726)

## Verify

- [x] Code runs without errors
- [x] Tests pass with 100% coverage

### CODE REVIEW

File: Migration.md [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+```
+
+The new TypeScript SDK uses an ES module import statement and configuration variable for the object containing your API key. This variable is used to instantiate a new configuration:
+
+```typescript
+import {
+  Configuration,
+  AddressesApi,
+  AddressEditable,
+} from "lob-typescript-sdk";
```
Comment (mwj8410) [inline Migration.md:40]: Let's remove AddressesApi and AddressEditable

These are not needed for the sample below

File: Migration.md [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+```
+
+​ _Note:_ If you are using the new TypeScript SDK using JavaScript here is the code that you would use for this:
+
+```javascript
+const {
+    AddressesApi,
+    AddressEditable,
+    Configuration
+} = require("@lob/lob-typescript-sdk");
```
Comment (mwj8410) [inline Migration.md:54]: Same as the TS portion. Only Configuration is needed

File: Migration.md [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+
+const config = new Configuration({
+    username: process.env.LOB_API_KEY
```
Comment (mwj8410) [inline Migration.md:57]: code requires closing `});`

File: Migration.md [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+```
+
+Like lob-node this approach uses Javascript built-in require with ES6 destructuring to extract the essential classes from the results of requiring the TypeScript SDK.
```
Comment (mwj8410) [inline Migration.md:60]: Typo: "Javascript" should be "Javascript's"

I'm also not sure we should say "Like lob-node". The essential sentence here is "this approach uses destructuring"

File: Migration.md [hunk 5]
```diff
@@ -0,0 +1,351 @@
...
+$ npm install lob
+```
+
+## IMPORT AND INITIALIZE
+
+Lob-node uses JavaScript’s built-in require. You initialize by passing your API key as an argument:
+
+```javascript
+const Lob = require("lob")("YOUR API KEY");
+​
```
Comment (mwj8410) [inline Migration.md:30]: the extra line is not needed

File: Migration.md [hunk 6]
```diff
@@ -0,0 +1,351 @@
...
+  console.error(err);
+}
+```
+
+### COMPARE LIST METHODS
+
+Here is a sample of a lob-node LIST method:
+
+```javascript
+Lob.addresses.list({ limit: 2 }, function (err, address) {
```
Comment (mwj8410) [inline Migration.md:133]: to be more correct, lets call "address" "addresses"

File: Migration.md [hunk 7]
```diff
@@ -0,0 +1,351 @@
...
+    return;
+  }
+});
+```
+
+Here is a sample of the TypeScript SDK LIST method:
+
+```typescript
+try {
+  const response = await new AddressesApi(config).list();
```
Comment (mwj8410) [inline Migration.md:145]: let's name this variable "addresses" as well

File: Migration.md [hunk 8]
```diff
@@ -0,0 +1,351 @@
...
+    return;
+  }
+});
+```
+
+Here is a sample of the TypeScript SDK DELETE method:
+
+```typescript
+try {
+  const deletedAddress = await new AddressesApi(config).delete("adr_xxxx");
```
Comment (mwj8410) [inline Migration.md:191]: lets use the same id between the two samples. So either "adr_43769b47aed248c2" or "adr_xxxx"

File: Migration.md [hunk 9]
```diff
@@ -0,0 +1,351 @@
...
+    zip_code: "07090",
+};
+const addressList={
+    addresses: [verificationData1, verificationData2]
+}
+
+try {
+  const bulkVerified = await UsVerifications.verifyBulk(addressList);
+} catch (err: any) {
+       console.error(err);
```
Comment (mwj8410) [inline Migration.md:251]: has extra level of indentation

File: Migration.md [hunk 10]
```diff
@@ -0,0 +1,351 @@
...
+    description: "Test Bank Account",
+    routing_number: "322271627",
+    account_number: "123456789",
+    signatory: "Gomez Addams",
+    account_type: BankTypeEnum.Individual,
+});
+
+let id = "";
+try {
+    const result = await new BankAccountsApi(config).create(bankData)
```
Comment (mwj8410) [inline Migration.md:321]: semicolon at the end of line

File: Migration.md [hunk 11]
```diff
@@ -0,0 +1,351 @@
...
+    description: "Newer Template",
+    html: "<html>Updated HTML for {{name}}</html>",
+});
+
+try {
+     const createTemplate = await Templates.create(templateData);
+     const updateData= new TemplateUpdate ({
+         description: "updated template",
+         published_version: createTemplate.published_version?.id as string
+     })
```
Comment (mwj8410) [inline Migration.md:347]: missing semicolon

File: Migration.md [hunk 12]
```diff
@@ -0,0 +1,351 @@
...
+     const updateTemplate = await Templates.update(createTemplate.id updateData);
+} catch (err: any) {
+     console.error(err);
```
Comment (mwj8410) [inline Migration.md:350]: need to close the catch statement

File: Migration.md [hunk 13]
```diff
@@ -0,0 +1,351 @@
...
+
+### TEMPLATE UPDATE METHOD
+
+The Template Update endpoint updates the description and/or published version of the template with a given id. This endpoint is not currently supported in lob-node. However, this is how it is done in the TypeScript SDK:
+
+```typescript
+const Templates = new TemplatesApi(config);
+const templateData: new TemplateWritable({
+    description: "Newer Template",
+    html: "<html>Updated HTML for {{name}}</html>",
```
Comment (mwj8410) [inline Migration.md:339]: remove trailing comma

File: Migration.md [hunk 14]
```diff
@@ -0,0 +1,351 @@
...
+const verificationData= new BankAccountVerify({
+    amounts: [11, 35],
+});
+
+const bankData= new BankAccountWritable = ({
+    description: "Test Bank Account",
+    routing_number: "322271627",
+    account_number: "123456789",
+    signatory: "Gomez Addams",
+    account_type: BankTypeEnum.Individual,
```
Comment (mwj8410) [inline Migration.md:316]: remove trailing comma (I see several other object that have trailing commas after the last attribute. My preference is to standardize on not using it

File: Migration.md [hunk 15]
```diff
@@ -0,0 +1,351 @@
...
+    state: "NJ",
+    zip_code: "07090"
+};
+const verificationData2= {
+    primary_line: "1515 CEMETERY LN",
+    city: "WESTFIELD",
+    state: "NJ",
+    zip_code: "07090",
+};
+const addressList={
```
Comment (mwj8410) [inline Migration.md:244]: spaces surrounding "="

File: Migration.md [hunk 16]
```diff
@@ -0,0 +1,351 @@
...
+} catch (err: any) {
+  console.error(err);
+}
+```
+
+​Again, as mentioned above, If you are using the new TypeScript SDK using JavaScript,the code is essentially the same apart from removing the type identifiers found in TypeScript This pattern may be followed for all methods where examples are not supplied as shown here:
+
+```javascript
+const UsVerifications = new USVerificationsApi(av_config);
+const verificationData1= {
```
Comment (mwj8410) [inline Migration.md:232]: space before "="

**Review by mwj8410 — CHANGES_REQUESTED** [review]
A few stylistic changes

File: Migration.md [hunk 17]
```diff
@@ -0,0 +1,285 @@
+# Migrate from Lob’s JavaScript to TypeScript SDK
+
+This guide illustrates differences between Lob’s current JavaScript library (lob-node) and our new TypeScript SDK (lob-typescript-sdk). The new TypeScript SDK supports all generally available endpoints and has the benefit of being built with TypeScript.
+We have also compared performance and found the new TypeScript SDK is equivalent or improved in most cases with some operations experiencing a consistent 2X improvement.
+In this guide we compare the following method pattern.
```
Comment (SidneyAllen) [inline Migration.md:5]: Add the following above "In this guide ..."

Note: apps written in JavaScript or TypeScript can both utilize Lob’s TypeScript SDK

Comment (SidneyAllen) [inline Migration.md:5]: Update line "In this guide ..." to 

In this guide we compare how lob-node and lob-typescript-sdk implement the following method pattern.

File: Migration.md [hunk 18]
```diff
@@ -0,0 +1,285 @@
...
+```
+The new TypeScript SDK uses an ES module import statement and configuration variable for the object containing your API key. This variable is used to instantiate a new configuration:
+```typescript
+import { Configuration } from "lob-typescript-sdk";
+
+const config: Configuration = new Configuration({
+  username: "<<YOUR API KEY HERE>>"
+});
+```
+_Note:_ If you are using the new TypeScript SDK using JavaScript here is the code that you would use for this:
```
Comment (SidneyAllen) [inline Migration.md:31]: Replace line "If you are using ..." with 

To use the new TypeScript SDK in a JavaScript app the code looks like this:

File: Migration.md [hunk 19]
```diff
@@ -0,0 +1,285 @@
...
+```javascript
+const { Configuration } = require("@lob/lob-typescript-sdk");
+
+const config = new Configuration({
+  username: process.env.LOB_API_KEY
+});
+```
+This approach uses ES6 destructuring to extract the essential classes from the results of requiring the TypeScript SDK..
+## Error Handling
+In the new Typescript SDK, errors returned by the API are thrown, not returned to an error first callback (See below examples in Compare Create methods). As a result, consumer code does not need to do the work of detecting an error, and can
```
Comment (SidneyAllen) [inline Migration.md:41]: The Error Handling section is missing an ending?  it says "detecting an error, and can". Sentence incompelte

**Review by SidneyAllen — CHANGES_REQUESTED** [review]
Real close - just a few suggestions

File: Migration.md (hunk 1 above)
Comment (1makedadavis) [reply Migration.md]: Done!

File: Migration.md (hunk 19 above)
Comment (1makedadavis) [reply Migration.md]: All revisions made.

**Pull request merged**